aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -86,33 +86,50 @@ private AWSDataStorePlugin(
     }
 
     /**
-     * Creates an {@link AWSDataStorePlugin} which can warehouse the model types provided by
+     * Constructs an {@link AWSDataStorePlugin} which can warehouse the model types provided by
+     * your application's code-generated model provider. This model provider is expected to have the
+     * fully qualified class name of com.amplifyframework.datastore.generated.model.AmplifyModelProvider.
+     *
+     * This constructor will attempt to find that provide by means of reflection. If you have changed
+     * the path to your {@link ModelProvider} instance, and do not use the default path generated by
+     * the Amplify Code Generator, then this will be break. Likewise, if you need to provide a custom
+     * {@link ModelProvider}, this will not work. In both cases, you should prefer one of the other
+     * overloads such as {@link #AWSDataStorePlugin(ModelProvider)}.
+     *
+     * If remote synchronization is enabled, it will be performed via {@link Amplify#API}.
```

Comment:
When I started looking at DataStore, this was one of the points of confusion for me. How do I tell if remote sync is enabled? It doesn't appear to be a setting in the category or plugin configuration, so I'm asserting that it's simply by virtue of an API being present. If that's the case, maybe we should say something like:

> If the API plugin is present and configured, then remote synchronization will be performed.

Or something along those lines

Reply:
Good point, done!